File: python/src/main/python/store/WaterIceCreamStore.py
from icecream.water.GuavaWaterIceCream import *
from icecream.water.MangoWaterIceCream import *
from icecream.water.PineappleWaterIceCream import *
from icecream.water.StrawberryWaterIceCream import *
from IceCreamStore import *


class WaterIceCreamStore(IceCreamStore):
    def createIceCream(self, selectedFlavor):
        if (selectedFlavor == "Strawberry"):
            return StrawberryWaterIceCream()
        elif (selectedFlavor == "Pineapple"):
            return PineappleWaterIceCream()
        elif (selectedFlavor == "Mango"):
            return MangoWaterIceCream()
        elif (selectedFlavor == "Guava"):
            return GuavaWaterIceCream()
        return None

File: java/src/main/java/mx/iteso/factory/stores/WaterIceCreamStore.java
package mx.iteso.factory.stores;

import mx.iteso.factory.IceCreamStore;
import mx.iteso.factory.icecreams.IceCream;
import mx.iteso.factory.icecreams.water.GuavaWaterIceCream;
import mx.iteso.factory.icecreams.water.MangoWaterIceCream;
import mx.iteso.factory.icecreams.water.PineappleWaterIceCream;
import mx.iteso.factory.icecreams.water.StrawberryWaterIceCream;

/** Water Ice Cream store class.*/
public class WaterIceCreamStore extends IceCreamStore {

    /**
     * Create ice cream.
     * @param selectedFlavor the flavor.
     * @return IceCream.
     */
    @Override
    protected IceCream createIceCream(final String selectedFlavor) {
        if (selectedFlavor.equals("Strawberry")) {
            return new StrawberryWaterIceCream();
        } else if (selectedFlavor.equals("Pineapple")) {
            return new PineappleWaterIceCream();
        } else if (selectedFlavor.equals("Mango")) {
            return new MangoWaterIceCream();
        } else if (selectedFlavor.equals("Guava")) {
            return new GuavaWaterIceCream();
        }
        return null;
    }
}
